src/components/layouts/pages/inspect/tree/Tree.js

```diff
@@ -146,20 +146,43 @@ function CheckList({ checks, selectedCheck, onCheckClick, errorsMap, ruleIndex }
         const checkKey = `${ruleIndex}:${context}`;
         const checkTitle = getCheckTitle({ context, index, allChecks: checksSorted, errorsMap, ruleIndex });
         return (
-            <ListItem
+            <LI
                 key={index}
                 onClick={() => onCheckClick(checkKey)}
                 selected={selectedCheck === checkKey}
                 button
                 className="check-item"
                 title={context}
-            >
-                <ListItemText primary={checkTitle} />
-            </ListItem>
+                checkTitle={checkTitle}
+            />
         );
     });
 }
 
+function LI({ selected, title, checkTitle, onClick, className }) {
+    // prevent auto scroll when was clicked itself
+    const [disableAutoScroll, setDisableAutoScroll] = useState(false);
+    const onItemClick = useCallback(() => {
+        setDisableAutoScroll(true);
+        return onClick();
+    }, [onClick]);
+    const listItem = useRef();
+    useEffect(() => {
+        if (selected && !disableAutoScroll) {
+            // To be sure that list expanded before auto scroll
+            setTimeout(() => listItem.current.scrollIntoView({ block: 'center' }), 0);
+        } else {
+            setDisableAutoScroll(false);
+        }
+    }, [selected]); // don't put disableAutoScroll in deps, it will make always to autoscroll
```

Comment:
It's possible to avoid console warnings by wrapping function inside `useEffect` into `useCallback`